backend/src/resolvers/users.resolver.ts

```diff
@@ -0,0 +1,70 @@
+import { SignupDto } from "../dtos/signup.dto";
+import UserDatasource from "../datasources/user.datasource";
+import { AuthenticationError, UserInputError } from "apollo-server";
+import { JwtPayload } from "../jwt-payload";
+import * as jwt from "jsonwebtoken";
+import { SigninDto } from "../dtos/signin.dto";
+import * as bcrypt from "bcrypt";
+import User from "../entities/user.entity";
+import PostDatasource from "../datasources/post.datasource";
+
+export const query = {
+  users: async (_source, _args, ctx) => {
+    const { userDatasource } = ctx.dataSources;
+
+    return userDatasource.getUsers();
+  },
+};
+
+export const properties = {
+  posts: async (parent: User, _args, ctx) => {
+    const postDatasource: PostDatasource = ctx.dataSources.postDatasource;
+    return postDatasource.getPostsByUser(parent.id);
+  },
+};
+
+export const mutation = {
+  // signup(name: String!, email: String!, password: String!): String
+  async signup(_, args: SignupDto, ctx) {
+    const { name, email, password } = args;
+    const userDatasource: UserDatasource = ctx.dataSources.userDatasource;
+    const existentUser = await userDatasource.getUserByEmail(email);
+
+    if (password.length < 8) {
+      throw new UserInputError("Password must at least 8 characters long");
```

Comment:
The user won't see this personalized message.

- **Why ?** Your resolver.ts include this mutation without catch of eventual errors so every **throw** here will results to an Not Authorized message for the user as your can see in your tests, which doesn't make sence. There are no authorization needed for signup and login
- **How to fix it** Return the error instead of throw or add a catch in your resolvers and return all error throws
```suggestion
      return new UserInputError("Password must at least 8 characters long");
```